### ISSUE TITLE
Syntax highlighting and brace matching for "switch/when" statements

### ISSUE THREAD

**Issue by robcowell** [forcedotcom/salesforcedx-vscode#2134]
### Summary

Syntax highlighting and brace matching doesn't correctly parse switch/when clauses

### Steps To Reproduce:

1.  Write a switch on/when structure as per screenshot.

### Expected result

1. Strings in when clause should be consistently syntax highlighted
1. Bracket matching should match up pairs

### Actual result

1. Odd colouring from the syntax highlighting
1. Need to put a trailing space between end of when clause and opening brace { even when the brace is on a new line or the brace pairs aren't matched both in syntax and on save/deploy

### Additional information

![image](https://user-images.githubusercontent.com/2143167/80120209-90cfc680-8582-11ea-9c15-6f42a5b5110e.png)

**VS Code Version**: 1.44.2

**SFDX CLI Version**: sfdx-cli/7.53.0-839fc82926 win32-x64 node-v10.15.3

**OS and version**: Win 10 x64 1909 build 18363.778

**Comment by zaynelt** [comment]
Hey Rob--just speaking to the syntax highlighting behavior, it doesn't actually look like you're using Codey Midnight (v 1.1.1) in this screenshot. Did you override any of the syntax styles on your local install?

**Comment by robcowell** [comment]
No, as far as I can tell, it's a stock install
![image](https://user-images.githubusercontent.com/2143167/80122349-54519a00-8585-11ea-9a25-ce6d377d0ac8.png)

**Comment by zaynelt** [comment]
Hmm. I do see some color oddities in your install--but, I also can repro the syntax highlighting issue you're seeing: 
![image](https://user-images.githubusercontent.com/16801980/80123009-2c414700-8543-11ea-8b6b-cb51bc510e74.png)

**Comment by robcowell** [comment]
Yup, I flipped themes a couple of times after seeing your post and it persists.  The brace matching/trailing space issue is more than just syntax highlighting too, it triggers VS Code's "Problems" tab to list an issue

**Comment by robcowell** [comment]
Colo(u)r oddities are probably due to my Bracket Pair Colorizer plugin I use to aid visual matching and ensuring I close out all pairs ;-)

**Comment by lcampos** [comment]
@robcowell @zaynelt this is not an issue with a theme but with the syntax highlighting rules we have for Apex. Looks like the use of `:` is throwing off the rule evaluation. We'll release a fix for it in the following releases.

**Comment by mohanraj-r** [comment]
I am able to reproduce this with v54.6.1 
![image](https://user-images.githubusercontent.com/1509984/161319883-933d2adf-0e94-4e15-aad5-f0f099407d89.png)

Seems also related to https://github.com/forcedotcom/salesforcedx-vscode/issues/967

**Comment by Stassy** [comment]
Same issue with "-" and inside siwtch statement not recognized:

![image](https://user-images.githubusercontent.com/6544017/231004584-a7ed01cf-cd2e-4722-9f16-666ff116d849.png)

**Comment by jeffjfreeman** [comment]
I'm also having the same issue on `v61.1.2`:

![image](https://github.com/forcedotcom/salesforcedx-vscode/assets/1069783/b50f2c2f-c264-4858-ad71-32e40db0e620)

**Comment by Stassy** [comment]
Problem still there with 2.87-7
Any solutions?

**Comment by TachyonicSpace** [comment]
I am still having the same issue on V64.7.1
